reduction_scripts/reduce_data.py

```diff
@@ -0,0 +1,848 @@
+                    f = pyfits.open(in_fn)
+                    sci_header = f[0].header
+                    sci_time = sci_header['DATE-OBS']
+                    # Now get the arc times
+                    arc_times = ['','']
+                    for i in range(2):
+                        # Fetch the arc time from the "extra" pkl file
+                        local_wsol_out_fn_extra = os.path.join(out_dir, '%s.wsol.fits_extra.pkl' % (local_arcs[i]))
+                        f = open(local_wsol_out_fn_extra, 'rb') 
+                        try:
+                            f_pickled = pickle.load(f, protocol=2) # TODO: see if this works
+                        except:
+                            f_pickled = pickle.load(f) # Python 3
+                        f.close()
+                        arc_times[i] = f_pickled[-1][0]
+                    
+                    # Now, make sure the Science is between the arcs:
+                    t0 = datetime.datetime( np.int(arc_times[0].split('-')[0]),
+                                            np.int(arc_times[0].split('-')[1]),
+                                            np.int(arc_times[0].split('-')[2].split('T')[0]),
+                                            np.int(arc_times[0].split('T')[1].split(':')[0]),
+                                            np.int(arc_times[0].split(':')[1]),
+                                            np.int(arc_times[0].split(':')[2].split('.')[0]),
+                                            )
+                    t1 = datetime.datetime( np.int(sci_time.split('-')[0]),
+                                            np.int(sci_time.split('-')[1]),
+                                            np.int(sci_time.split('-')[2].split('T')[0]),
+                                            np.int(sci_time.split('T')[1].split(':')[0]),
+                                            np.int(sci_time.split(':')[1]),
+                                            np.int(sci_time.split(':')[2].split('.')[0]),
+                                            )
+                    t2 = datetime.datetime( np.int(arc_times[1].split('-')[0]),
+                                            np.int(arc_times[1].split('-')[1]),
+                                            np.int(arc_times[1].split('-')[2].split('T')[0]),
+                                            np.int(arc_times[1].split('T')[1].split(':')[0]),
+                                            np.int(arc_times[1].split(':')[1]),
+                                            np.int(arc_times[1].split(':')[2].split('.')[0]),
+                                            )
+                    ds1 = (t1 - t0).total_seconds()
+                    ds2 = (t2 - t1).total_seconds()
+                    if ds1>0 and ds2>0:
+                        # Alright, I need to interpolate betweent the two arcs
+                        w1 = ds2/(ds1+ds2)
+                        w2 = ds1/(ds1+ds2)
```

Comment:
Indeed! This is something I've been looking at but I'm not 100% sure if it really makes any difference. I guess we never decided anything about but neither solved it. I've added a condition within the function just in case as it's shown in the image.

<img width="503" alt="Screenshot 2023-11-14 at 10 37 09 pm" src="https://github.com/PyWiFeS/pipeline/assets/44895424/9ff235a6-a303-4b8b-acd2-8934fbf5b0af">

Reply:
Best to use `else` rather than a second `if`. That or if, elif, with an else at the end to throw an error. But I think we're generally assuming that the data in the headers in valid.

Besides that, sounds good to me. Worst case, it's redundant.

Reply:
Ok, done! Thank you!

Reply:
Done. See comment below in new commitment https://github.com/PyWiFeS/pipeline/pull/29#discussion_r1393657900

Reply:
Ah, I would actually have done `else` rather than `elif file_camera == 'WiFeSBlue':`. ('if elif' implies that 'else' is valid, just unhandled, same reason I wouldn't do 'if if')

It's an ok case for a comment if you'd like each case to be clearly labelled:

```py
                        if file_camera == 'WiFeSRed':
                            w1 = ds1/(ds1+ds2)
                            w2 = ds2/(ds1+ds2)
                        else: # file_camera == 'WiFeSBlue'
                            w1 = ds2/(ds1+ds2)
                            w2 = ds1/(ds1+ds2)
```

Reply:
Ok, I now understand what you meant! Thank you. Done.